opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1506,95 +1606,422 @@ namespace Opm
 
 
 
-
     template <typename TypeTag>
     void
     MultisegmentWell<TypeTag>::
-    assembleControlEq(Opm::DeferredLogger& deferred_logger) const
+    assembleControlEq(const WellState& well_state, const Opm::Schedule& schedule, const SummaryState& summaryState, Opm::DeferredLogger& deferred_logger)
     {
         EvalWell control_eq(0.0);
 
-        switch (well_controls_get_current_type(well_controls_)) {
-            case THP: // not handling this one for now
+        const auto& well = well_ecl_;
+        const int well_index = index_of_well_;
+        double efficincyFactor = well.getEfficiencyFactor();
+
+        if (wellIsStopped_) {
+            control_eq = getSegmentGTotal(0);
+            return;
```

Comment:
With this return, which is not present in the StandardWell::assembleControlEq(), we do not change `resWell_` or `duneD_`. Is that correct?

Reply:
This is a bug. Thanks for noticing. It is correct for standard well.